### ISSUE TITLE
Creating LB with existing GKE cluster fails with error about instance group membership

### ISSUE THREAD

**Issue by mmllc-jsilverman** [terraform-google-modules/terraform-google-lb-http#322]
### TL;DR

I have an existing cluster; the cluster is 3 nodes. There are existing Manage Instance Groups and Unmanaged Instance Groups.

I want to be able to create a new Load Balancer using this module, rather than via a Kubernetes Ingress resource. 

I am passing in a list of InstanceGroup URLs as the value to backends.groups[] in this module and I get an error.

```
Error: Error waiting to create BackendService: Error waiting for Creating BackendService:
Validation failed for instance 'projects/my-project/zones/us-central1-a/instances/gke-microservices-0-microservices-0-3b1f2fe25b31-38b1':
instance may belong to at most one load-balanced instance group

```

### Expected behavior

Terraform creates the LB, and backend, and adds these nodes to the backend

### Observed behavior

The following error occurs:

```
Error: Error waiting to create BackendService: Error waiting for Creating BackendService:
Validation failed for instance 'projects/my-project/zones/us-central1-a/instances/gke-microservices-0-microservices-0-3b1f2fe25b31-38b1':
instance may belong to at most one load-balanced instance group

```

The Managed Instance Groups have the following URL format:
```
"https://www.googleapis.com/compute/v1/projects/my-project/zones/us-central1-a/instanceGroupManagers/gke-microservices-0-microservices-0-c5a08770-grp",
```
After reading quite a bit, I've found that the problem might be "Managed" Instance Groups; I can't add any given instance to more than one MIG

So I munged the URLs to this format. This also better follows the example as given in this repository:

```
"https://www.googleapis.com/compute/v1/projects/my-project/zones/us-central1-a/instanceGroups/gke-microservices-0-microservices-0-c5a08770-grp",
```

However, I am still getting this error.

It seems as though this LB module is trying to create a new Managed Instance Group and add it to the balancer backend. However, that doesn't make sense based on the docs and the usage. One has to provide a list of Groups to the backend. Presumably these are existing Groups?

My LB Module implementation looks like this:

```
module "api_load_balancer_ext" {
  project                 = var.project_id
  source  = "GoogleCloudPlatform/lb-http/google"
  version = "= 8.0.0"
  name              = "api-ext"
  ssl               = true
  use_ssl_certificates   = true
  ssl_certificates       = [
    data.google_compute_ssl_certificate.api.self_link
  ]
  firewall_networks = [ module.gke-private-cluster-0.network_name ]
  load_balancing_scheme = "EXTERNAL"
  address = google_compute_global_address.api_ingress_ipv4_ext.address
  ipv6_address = google_compute_global_address.api_ingress_ipv6_ext.address
  target_tags = [local.target_tags]
  backends = {
    default = {
      description                     = null
      protocol                        = "HTTP"
      port                            = local.external_service_port
      port_name                       = local.external_service_port_name
      timeout_sec                     = 10
      connection_draining_timeout_sec = null
      enable_cdn                      = false
      compression_mode                = null
      security_policy                 = null
      session_affinity                = null
      affinity_cookie_ttl_sec         = null
      custom_request_headers          = null
      custom_response_headers         = null
      health_check = {
        check_interval_sec  = null
        timeout_sec         = null
        healthy_threshold   = null
        unhealthy_threshold = null
        request_path        = "/health/"
        port                = local.external_service_port
        host                = null
        logging             = true
      }
      log_config = {
        enable      = true
        sample_rate = 1.0
      }
      iap_config = {
        enable               = false
        oauth2_client_id     = null
        oauth2_client_secret = null
      }
      groups = [ for i in module.gke-private-cluster-0.cluster_instance_group_urls :
          {
            group = replace(i, "instanceGroupManagers", "instanceGroups")
            balancing_mode               = null
            capacity_scaler              = null
            description                  = null
            max_connections              = null
            max_connections_per_instance = null
            max_connections_per_endpoint = null
            max_rate                     = null
            max_rate_per_instance        = null
            max_rate_per_endpoint        = null
            max_utilization              = null
          }
      ]
    }
  }
}

```

The output of `module.gke-private-cluster-0.cluster_instance_group_urls` is a list of the first URL example I gave of instanceGroupManagers. There are three -- one per GKE node. Which leads me to a secondary question about the documentation -- in the example, only one Group was provided. But that didn't make sense to me since there are multiple instance groups for this backend. Is my dynamic `groups` block the correct approach?

Please advise

### Terraform Configuration

```hcl
I'm not entirely sure what to include here, apologies!

I'm using TF Cloud to plan and apply.
```

### Terraform Version

```sh
$ terraform version
Terraform v1.3.7
on linux_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v2.27.0
+ provider registry.terraform.io/hashicorp/google v4.61.0
+ provider registry.terraform.io/hashicorp/google-beta v4.61.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.19.0
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.1.3
```

**Comment by imrannayer** [comment]
@mmllc-jsilverman can you try [dynamic backend](https://github.com/terraform-google-modules/terraform-google-lb-http/tree/master/modules/dynamic_backends) sub-module?
